fedot/core/operations/evaluation/operation_implementations/models/boostings_implementations.py

```diff
@@ -92,21 +93,23 @@ def plot_feature_importance(self, importance_type='weight'):
         plot_feature_importance(features_names, model_output.values())
 
     @staticmethod
-    def convert_to_dataframe(data: Optional[InputData], identify_cats: bool):
-        dataframe = pd.DataFrame(data=data.features)
-        if data.target is not None and data.target.size > 0:
-            dataframe['target'] = np.ravel(data.target)
+    def convert_to_dataframe(input_data: Optional[InputData], identify_cats: bool):
+        copied_input_data = deepcopy(input_data)
+
+        dataframe = pd.DataFrame(data=copied_input_data.features)
+        if copied_input_data.target is not None and copied_input_data.target.size > 0:
+            dataframe['target'] = np.ravel(copied_input_data.target)
         else:
             # TODO: temp workaround in case data.target is set to None intentionally
             #  for test.integration.models.test_model.check_predict_correct
-            dataframe['target'] = np.zeros(len(data.features))
+            dataframe['target'] = np.zeros(len(copied_input_data.features))
 
-        if identify_cats and data.categorical_idx is not None:
-            for col in dataframe.columns[data.categorical_idx]:
+        if identify_cats and copied_input_data.categorical_idx is not None:
+            for col in dataframe.columns[copied_input_data.categorical_idx]:
                 dataframe[col] = dataframe[col].astype('category')
 
-        if data.numerical_idx is not None:
-            for col in dataframe.columns[data.numerical_idx]:
+        if copied_input_data.numerical_idx is not None:
+            for col in dataframe.columns[copied_input_data.numerical_idx]:
```

Comment:
why do we need additional copying of `InputData` instance?

Reply:
The reason for adding an additional copy of the `InputData` instance is to avoid unintended mutations. Currently, within that function, the `input_data` parameter is modified so that the `features` field ends up containing both feature values and `target` values. This causes issues later, particularly during the prediction phase, where these unintended modifications lead to failures.

To address this, we create a copy of `input_data` to preserve the original data and prevent any side effects. Here's an illustration of the problem:

![image](https://github.com/user-attachments/assets/6886f798-446b-4eba-b1a6-e54f858890cb)

I also tried an alternative approach:
```
target = dataframe["target"]
return dataframe.drop(columns=["target"], inplace=True), target
```
However, this approach led to runtime errors, likely due to issues with dynamic memory allocation when using `inplace=True` with `drop`. To ensure stability, I opted to stick with the copy-based solution.